### PR TITLE
Wenshu

### PR DESCRIPTION
**Pull request by Wenshu0206**
hw3test wenshu

### CODE REVIEW

File: hw3-test/test_hw3_wenshu0206.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    category: one-shot test
+    """
+    name = "my-django-app"
+    result = classify_by_name(name)
+
+    assert result == "Python"
+    assert result != "C++"
+    assert isinstance(result, str)
+
+def test_one_shot_known_language_mapping():
```
Comment (thomas0829) [inline hw3-test/test_hw3_wenshu0206.py]: This one-shot test is correct in concept - testing a known input-output pair. ✅

However, the second assertion is redundant:
```python
assert result == "Python"  # This already guarantees result is "Python"
assert result in ["Python", "C++", ...]  # This is redundant - if result == "Python", it's obviously in the list

File: hw3-test/test_hw3_wenshu0206.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Tests for classify_by_name()
+"""
+
+import sys
+from pathlib import Path
+sys.path.insert(0, str(Path(__file__).parent.parent))
+
+from utils.classify_languages import classify_by_name
+
+def test_classify_smoke_multiple_inputs():
```
Comment (thomas0829) [inline hw3-test/test_hw3_wenshu0206.py:15]: This test is checking specific output values (`results[0] == "Other"`, `results[3] == "TypeScript"`), 
which makes it more like a **one-shot test** rather than a **smoke test**.

A smoke test should only verify:
1. The function doesn't crash
2. It returns the expected type

Suggested fix:
```python
def test_classify_smoke_multiple_inputs():
    """
    author: Wenshu0206
    reviewer: thomas0829
    category: smoke test
    Smoke test: ensure function runs on multiple simple names without crashing.
    """
    names = ["repo", "python-utils", "opencv-lib", "vscode"]
    results = [classify_by_name(n) for n in names]
    
    # Only check type and basic structure
    assert all(isinstance(r, str) for r in results)
    assert len(results) == 4

File: hw3-test/test_hw3_wenshu0206.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    """
+    numeric = classify_by_name("123456789")
+    long_name = "python" + ("x" * 2000)
+    long_result = classify_by_name(long_name)
+
+    assert numeric == "Other"
+    assert long_result == "Python"
+    assert len(long_name) > 2000
+
+def test_pattern_tensorflow_priority():
```
Comment (thomas0829) [inline hw3-test/test_hw3_wenshu0206.py:76]: This test is checking a single specific case ("tensorflow-utils" → "Python"), which makes it a **one-shot test**, not a **pattern test**.

A pattern test should verify a **pattern holds across multiple cases**. For example:

```python
def test_pattern_tensorflow_priority():
    """
    author: Wenshu0206
    reviewer: thomas0829
    category: pattern test
    Pattern: names containing 'tensorflow' should consistently map to Python,
    regardless of other keywords, due to priority ordering.
    """
    tensorflow_names = [
        "tensorflow-utils",
        "tensorflow-models", 
        "my-tensorflow-project",
        "tensorflow-cpp",  # Even with 'cpp', should still be Python
        "tensorflow"
    ]
    
    results = [classify_by_name(name) for name in tensorflow_names]
    
    # Pattern: ALL should map to Python due to tensorflow keyword priority
    assert all(r == "Python" for r in results)

**Review by thomas0829 — CHANGES_REQUESTED** [review]

File: hw3-test/test_hw3_wenshu0206.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    reviewer: Muwen320
+    category: one-shot test
+
+    Known language mapping: system-design-primer → Python.
+    """
+    result = classify_by_name("system-design-primer")
+
+    assert result == "Python"
+    assert result in ["Python", "C++", "Other", "TypeScript", "JavaScript", "C#"]
+
```
Comment (Muwen320) [inline hw3-test/test_hw3_wenshu0206.py]: “Good use of a known repo name. The expected output is clear, and the test matches one-shot design. Nice work!”

File: hw3-test/test_hw3_wenshu0206.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    reviewer: Muwen320
+    category: one-shot test
+    """
+    name = "my-django-app"
+    result = classify_by_name(name)
+
+    assert result == "Python"
+    assert result != "C++"
+    assert isinstance(result, str)
+
```
Comment (Muwen320) [inline hw3-test/test_hw3_wenshu0206.py:44]: “This is a clear and effective one-shot test.
The input my-django-app is a representative Python-related repo name, and the test correctly checks a single expected behavior. Assertions are straightforward and focused, which matches the one-shot design requirement. Nice work!”

File: hw3-test/test_hw3_wenshu0206.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    category: edge test
+    """
+    numeric = classify_by_name("123456789")
+    long_name = "python" + ("x" * 2000)
+    long_result = classify_by_name(long_name)
+
+    assert numeric == "Other"
+    assert long_result == "Python"
+    assert len(long_name) > 2000
+
```
Comment (Muwen320) [inline hw3-test/test_hw3_wenshu0206.py:74]: “Great edge test.
This test covers two important edge scenarios: a purely numeric repo name and an extremely long Python-prefixed name. Both cases are realistic corner inputs, and the expected outputs (Other and Python) are consistent with how the classifier should behave.

The test is focused, meaningful, and clearly demonstrates that classify_by_name handles boundary conditions correctly. Nicely done.”

**Review by Muwen320 — DISMISSED** [review]
All tests look correct and align with the required categories. Approved!

**Review by thomas0829 — APPROVED** [review]
Great work on the test refactoring! All three changes demonstrate solid testing principles:

1. **Smoke test refinement** - Correctly removed specific value assertions, keeping only type and count checks. This aligns perfectly with smoke test philosophy: verify basic functionality without deep validation.

2. **Removing redundant assertions** - Eliminated the unnecessary `result in [...]` check in the one-shot test. If `result == "Python"` passes, the membership check is guaranteed to pass.

3. **Enhanced pattern test coverage** - Expanded from 1 test case to 5, including edge cases like `tensorflow-cpp` to validate priority logic. Removed redundant `!= "C++"` and `isinstance()` checks that were implicitly covered by `== "Python"`.

Overall, the tests are now cleaner, more comprehensive, and follow DRY principles.
